util.py

```diff
@@ -0,0 +1,108 @@
+                else:
+                    params.append(arg)
+
+            return await func(self, ctx, flags, params)
+        return wrapped
+    return wrapper
+
+# Yields all extension files in path.
+#   - path contains path to extensions                      default is 'extensions'
+#   - prefix_path toggles prefixing with extension path     default is False
+#   - recursive toggles recursive search                    default is True
+
+def yield_extensions(path: str = 'extensions', prefix_path: bool = False, recursive: bool = True) -> Generator[str, None, None]:
+    path = join(path, '**\\*.py' if recursive else '*.py')             # Build path dependent on requirements
+    for file in iglob(path, recursive=recursive):                      # Use iglob to match all python files
+        components = regex.findall(r'\w+', file)[:-1]                  # Split into components and trim extension
+        yield '.'.join(components) if prefix_path else components[-1]  # Either return import path or extension name
+
+# Finds extension in path, returns full extension path if found
+#   - extension contains extension to search for
+#   - path contains path to extensions                      default is 'extensions'
+#   - recursive toggles recursive search                    default is True
+
+def extension_path(extension: str, path: str = 'extensions', recursive: bool = True) -> str:
+    path = join(path, '**' if recursive else '', f'{extension_name(extension)}.py')  # Build path dependent on requirement
+    for file in iglob(path, recursive=recursive):                                    # Use iglob to match all python files
+        components = regex.findall(r'\w+', file)[:-1]                                # Split into components and trim extension
+        return '.'.join(components)                                                  # Return full extension path
+    return extension                                                                 # If not found return extension
+
+# Returns extension name from extension path
+#   - extension_path contains path to extension with `.` seperation
+
+def extension_name(extension_path: str) -> str:
+    return extension_path.split('.')[-1]
+
+# Returns default, empty embed.
+#   - title & description are header strings                default is empty
+#   - author toggles author                                 default is False
+#   - footer toggles footer                                 default is True
+#   - color loops through rainbow color palette             default is red
+
+def default_embed(bot: commands.Bot, title: str = '', description: str = '', author: bool = False, footer: bool = True, color: int = 0) -> discord.Embed:
+    palette = [
+        discord.Colour.from_rgb(255, 89,  94 ),
+        discord.Colour.from_rgb(255, 202, 58 ),
+        discord.Colour.from_rgb(138, 201, 38 ),
+        discord.Colour.from_rgb(25,  130, 196),
+        discord.Colour.from_rgb(106, 76,  147)
+    ]
```

Comment:
please add comments for each colour so I know which is which

Reply:
done